sql/mysqld.cc

```diff
@@ -8820,6 +8820,28 @@ static int get_options(int *argc_ptr, char ***argv_ptr)
   return 0;
 }
 
+/*
+  This function uses the maturity codes from _ and returns the respective maturity string as in
+  -unknown  -experimental -alpha -beta -gamma 
+  Note we do not display anything for stable versions*/
+  
+  const char* get_server_maturity_string(int maturity_code){
+    switch (maturity_code)
+    {
+    case 0: return "-unknown";
+            break;
+    case 1: return "-experimental";
+            break;  
+    case 2: return "-alpha";
+            break;
+    case 3: return "-beta";
+            break;
+    case 4: return "-gamma";
+            break;
+    default:return NULL;
```

Comment:
NULL will get interpreted as the last part when consumed by `strxnmov` so maybe `""` here (on a new line too).

Reply:
could remove the `SERVER_MATURITY_LEVEL` fixed argument too and `compile_assert` to ensure the prescribed level SERVER_MATURITY_LEVEL hasn't been altered.